src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -458,9 +459,12 @@ class ConstantPropagation extends Transform with DependencyAPIMigration with Res
     case _ => constPropMuxCond(m)
   }
 
-  private def constPropNodeRef(r: WRef, e: Expression) = e match {
-    case _: UIntLiteral | _: SIntLiteral | _: WRef => e
-    case _ => r
+  private def constPropNodeRef(r: WRef, e: Expression): Expression = {
+    def doit(ex: Expression) = ex match {
+      case _: UIntLiteral | _: SIntLiteral | _: WRef => ex
+      case _ => r
+    }
```

Comment:
![starsky-and-hutch](https://user-images.githubusercontent.com/1018530/87616554-4f633900-c6e3-11ea-8653-2657389abef4.gif)

Reply:
Hahaha exactly. I wasn't sure what to call the new inner function since it's not a `loop`, nor is it `rec`ursive. I've seen `doit` used before as a throwaway function name so 🤷